cmd/kube-proxy/app/server.go

```diff
@@ -465,6 +467,13 @@ addon that provides cluster DNS for these cluster IPs. The user must create a se
 with the apiserver API to configure the proxy.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			verflag.PrintAndExitIfRequested()
+
+			// Config and flags parsed, now we can initialize logging.
+			logs.InitLogs()
+			if err := logsapi.ValidateAndApplyAsField(&opts.config.Logging, utilfeature.DefaultFeatureGate, field.NewPath("logging")); err != nil {
```

Comment:
I don't think this is the right place to do this... until you run `Complete()` below, you haven't actually loaded the config file, so no log options in the config will be read yet here, and no feature gates values will have been loaded from the config file either

moving this after `Complete()` would mean the config would be loaded and feature gates from the config set

Reply:
Thanks, updated. 

I build a kube-proxy binary in my local to make a test, I passed `--logging-format json` and started it, but the start fails and returns："initialize logging: format: Invalid value: \"json\": Unsupported log format"

Is there anywhere else I missed?

```shell
➜  kubernetes git:(loggin_format) ✗ ./_output/local/bin/darwin/amd64/kube-proxy -v 1 --logging-format json  
I0509 22:10:49.284902   65090 server.go:235] "Warning, all flags other than --config, --write-config-to, and --cleanup are deprecated, please begin using a config file ASAP"
I0509 22:10:49.285015   65090 server_others.go:75] "Using iptables proxy"
I0509 22:10:49.285094   65090 log.go:194] {json 5s 1 [] {{false {{{0 0} {<nil>} 0 DecimalSI}}}}}
E0509 22:10:49.285133   65090 run.go:74] "command failed" err="initialize logging: format: Invalid value: \"json\": Unsupported log format"
```

Reply:
json logformat registered here: https://github.com/kubernetes/kubernetes/blob/e865b30abd0a862697ff1d2526ea8897ee98a523/staging/src/k8s.io/component-base/logs/json/register/register.go#L26

Reply:
You need to activate that code by importing the package:
```patch
diff --git a/cmd/kube-proxy/proxy.go b/cmd/kube-proxy/proxy.go
index 86ef9dff68e..085cc2e9f39 100644
--- a/cmd/kube-proxy/proxy.go
+++ b/cmd/kube-proxy/proxy.go
@@ -20,6 +20,7 @@ import (
        "os"
 
        "k8s.io/component-base/cli"
+       _ "k8s.io/component-base/logs/json/register"            // for JSON support
        _ "k8s.io/component-base/metrics/prometheus/restclient" // for client metric registration
        _ "k8s.io/component-base/metrics/prometheus/version"    // for version metric registration
        "k8s.io/kubernetes/cmd/kube-proxy/app"
```

Reply:
Thanks! it works now 😄